{
   "containers": {
      "cna": {
         "providerMetadata": {
            "orgId": "f4215fc3-5b6b-47ff-a258-f7189bd81038"
         },
         "descriptions": [
            {
               "lang": "en",
               "value": "In the Linux kernel, the following vulnerability has been resolved:\n\nfutex: Use correct exit on failure from futex_hash_allocate_default()\n\ncopy_process() uses the wrong error exit path from futex_hash_allocate_default().\nAfter exiting from futex_hash_allocate_default(), neither tasklist_lock\nnor siglock has been acquired. The exit label bad_fork_core_free unlocks\nboth of these locks which is wrong.\n\nThe next exit label, bad_fork_cancel_cgroup, is the correct exit.\nsched_cgroup_fork() did not allocate any resources that need to freed.\n\nUse bad_fork_cancel_cgroup on error exit from futex_hash_allocate_default()."
            }
         ],
         "affected": [
            {
               "product": "Linux",
               "vendor": "Linux",
               "defaultStatus": "unaffected",
               "repo": "https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git",
               "programFiles": [
                  "kernel/fork.c"
               ],
               "versions": [
                  {
                     "version": "7c4f75a21f636486d2969d9b6680403ea8483539",
                     "lessThan": "f1635765cd0fdbf27b04d9a50be91a01b5adda13",
                     "status": "affected",
                     "versionType": "git"
                  },
                  {
                     "version": "7c4f75a21f636486d2969d9b6680403ea8483539",
                     "lessThan": "4ec3c15462b9f44562f45723a92e2807746ba7d1",
                     "status": "affected",
                     "versionType": "git"
                  }
               ]
            },
            {
               "product": "Linux",
               "vendor": "Linux",
               "defaultStatus": "affected",
               "repo": "https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git",
               "programFiles": [
                  "kernel/fork.c"
               ],
               "versions": [
                  {
                     "version": "6.16",
                     "status": "affected"
                  },
                  {
                     "version": "0",
                     "lessThan": "6.16",
                     "status": "unaffected",
                     "versionType": "semver"
                  },
                  {
                     "version": "6.16.10",
                     "lessThanOrEqual": "6.16.*",
                     "status": "unaffected",
                     "versionType": "semver"
                  },
                  {
                     "version": "6.17",
                     "lessThanOrEqual": "*",
                     "status": "unaffected",
                     "versionType": "original_commit_for_fix"
                  }
               ]
            }
         ],
         "cpeApplicability": [
            {
               "nodes": [
                  {
                     "operator": "OR",
                     "negate": false,
                     "cpeMatch": [
                        {
                           "vulnerable": true,
                           "criteria": "cpe:2.3:o:linux:linux_kernel:*:*:*:*:*:*:*:*",
                           "versionStartIncluding": "6.16",
                           "versionEndExcluding": "6.16.10"
                        },
                        {
                           "vulnerable": true,
                           "criteria": "cpe:2.3:o:linux:linux_kernel:*:*:*:*:*:*:*:*",
                           "versionStartIncluding": "6.16",
                           "versionEndExcluding": "6.17"
                        }
                     ]
                  }
               ]
            }
         ],
         "references": [
            {
               "url": "https://git.kernel.org/stable/c/f1635765cd0fdbf27b04d9a50be91a01b5adda13"
            },
            {
               "url": "https://git.kernel.org/stable/c/4ec3c15462b9f44562f45723a92e2807746ba7d1"
            }
         ],
         "title": "futex: Use correct exit on failure from futex_hash_allocate_default()",
         "x_generator": {
            "engine": "bippy-1.2.0"
         }
      }
   },
   "cveMetadata": {
      "assignerOrgId": "f4215fc3-5b6b-47ff-a258-f7189bd81038",
      "cveID": "CVE-2025-39976",
      "requesterUserId": "gregkh@kernel.org",
      "serial": "1",
      "state": "PUBLISHED"
   },
   "dataType": "CVE_RECORD",
   "dataVersion": "5.0"
}
